### PR TITLE
lib/ukargparse: Add basic tests

### PR DESCRIPTION
**Pull request by cryeprecision**
### Prerequisite checklist

 - [X] Read the [contribution guidelines](https://docs.unikraft.org/contribute.html) regarding submitting new changes to the project;
 - [X] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.pl`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.pl) on your commit series before opening this PR;
 - [X] Updated relevant documentation.

### Additional configuration

 - `CONFIG_LIBUKARGPARSE_TESTS=y`
 - `CONFIG_LIBUKARGPARSE_TESTS_PARSE=y`
 - `CONFIG_LIBUKARGPARSE_TESTS_NULL=y`
 - `CONFIG_LIBUKARGPARSE_TESTS_EXTRA=y`

### Description of changes

Added some basic tests for the `ukargparse`-library.

### CODE REVIEW

File: lib/ukargparse/tests/test_ukargparse_parse.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Simple memory pool using LIFO principle
+ *
+ * Authors: Rico Muench <muench.rico@gmail.com>
+ *
+ *
+ * Copyright (c) 2020, NEC Laboratories Europe GmbH, NEC Corporation,
+ *                     All rights reserved.
```
Comment (nderjung) [inline lib/ukargparse/tests/test_ukargparse_parse.c:9]: You can contribute the copyright to yourself, it does not necessarily need to be NEC :)

**Review by nderjung — CHANGES_REQUESTED** [review]
Hi @cryeprecision, thanks for PR and the great work at [USoC21](https://usoc21.unikraft.org)!

To get this PR ready to be merged, please can you:

 * Remove the `uktest` base commits (4f3bb8c through 8ef440e) as these will be introduced via separate PRs and so will cause a merge conflict;
  * Please check the results of the `checkpatch.pl` script, there seem to be errors which need addressing.

Thanks again!

Depends-on: #218

**Comment by razvand** [comment]
@cryeprecision , can you take at look at this PR and make the requested fixes?

**Comment by razvand** [comment]
This will be closed, since it has been resubmitted by @StefanJum in #392

**Pull request closed without merging**